lib/state/ui/middleware.ts

```diff
@@ -39,7 +48,46 @@ export default store => {
         }
         break;
     }
-
+    switch (action.type) {
+      case 'App.closeNote':
+      case 'App.showAllNotes':
+      case 'App.selectTrash':
+      case 'App.selectTag':
+        store.dispatch(setSelectedNote(null));
+        break;
+      case 'App.notesLoaded':
+        if (store.getState().ui.note) {
+          store.dispatch(
+            setSelectedNote(
+              action.notes.length
+                ? action.notes.find(
+                    note => note.id === store.getState().ui.note.id
+                  )
+                : store.getState().ui.note
+            )
+          );
+        } else {
+          store.dispatch(
+            setSelectedNote(action.notes.length ? action.notes[0] : null)
+          );
```

Comment:
similarly we can move this into the reducer as well since `action.notes` is available to us in the reducer

```ts
const setSelectedNote = (
   state = defaultNote,
   { type, note }: { type: string; note: T.NoteEntity[] }
 ) => {
  switch (type ) {
    …
    case 'App.notesLoaded': {
      if ( ! state ) {
        return action.notes[0] || null;
      }

      const note = action.notes.find( ( { id } ) => id === state.id );
      return note || state;
    }
    …
}
```

fun fact, although the entire `switch` block shares variable name context, if we add curlies to a `case` then we get a new context and can create variable names without conflict with other `case` blocks

Reply:
done